File: leetcode-solutions/greedy/solution-2554/solution.js
/**
 * @param {number[]} banned
 * @param {number} n
 * @param {number} maxSum
 * @return {number}
 */
var maxCount = function(banned, n, maxSum) {
  banned = new Set(banned);
  let sum = 0;
  let res = 0;
  for (let i = 1; i <= n; i++) {
    if (banned.has(i)) continue;
    if (sum + i <= maxSum) {
      res++;
      sum += i;
    } else break;
  }
  return res;
};

File: leetcode-solutions/greedy/solution-2554/solution.go
package solution2554

func maxCount(banned []int, n int, maxSum int) (res int) {
    bannedMap := make(map[int]bool)
    for i := 0; i < len(banned); i++ {
        bannedMap[banned[i]] = true
    }
    sum := 0
    for i := 1; i <= n; i++ {
        if _, ok := bannedMap[i]; ok {
            continue
        }
        if sum+i <= maxSum {
            res++
            sum += i
        } else {
            break
        }
    }
    return
}
